cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -161,16 +168,38 @@ public CruiseControlApiImpl(Vertx vertx) {
                 .get(port, host, path, response -> {
                     response.exceptionHandler(result::fail);
                     if (response.statusCode() == 200 || response.statusCode() == 201) {
-                        String userTaskID = response.getHeader(USER_ID_HEADER);
+                        String userTaskID = response.getHeader(CC_REST_API_USER_ID_HEADER);
                         response.bodyHandler(buffer -> {
-                            JsonObject json = buffer.toJsonObject();
+                            JsonObject jsonUserTask = buffer.toJsonObject().getJsonArray("userTasks").getJsonObject(0);
+                            JsonObject json = new JsonObject()
+                                    .put("Status", jsonUserTask.getString("Status"))
+                                    .put("summary", ((JsonObject) Json.decodeValue(jsonUserTask.getString("originalResponse"))).getJsonObject("summary"));
+
                             if (json.containsKey(CC_REST_API_ERROR_KEY)) {
                                 result.fail(json.getString(CC_REST_API_ERROR_KEY));
                             } else {
                                 CruiseControlResponse ccResponse = new CruiseControlResponse(userTaskID, json);
                                 result.complete(ccResponse);
                             }
                         });
+                    } else if (response.statusCode() == 500) {
```

Comment:
I've never see a 500 Error on the `user_tasks` endpoint? Have you? I agree we should def cover it, I just wonder what would trigger it?

Reply:
Maybe I was wrong here because at the beginning the user tasks endpoint was used during rebalance proposal and it gave me the 500 error. After switching to handle rebalance proposal polling on the rebalance endpoint only, maybe I forgot to revert back here. If you have never seen a 500 error for user tasks then it makes sense to revert back. Did you never seen this because you never had an error on status tasks endpoint or because you saw errors but status code was 200?